run.sh

```diff
@@ -1,15 +1,71 @@
-  devpi use http://localhost:3141
-  devpi login root --password=''
-  devpi user -m root password="${DEVPI_PASSWORD}"
+
+
+#===========================
+# Default environment if none 
+# is provided
+#===========================
+DEFAULT_DEVPI_SERVERDIR=/devpidata
+DEFAULT_NGINX_IP=0.0.0.0
+DEFAULT_NGINX_PORT=3141
+DEFAULT_DEVPI_SERVERIP=127.0.0.1
+DEFAULT_DEVPI_SERVERPORT=3142
+DEFAULT_DEVPI_CLIENTDIR=/tmp/devpi-client
+DEFAULT_DEVPI_CLIENTNAME=root
+DEFAULT_DEVPI_CLIENTPWD=
+DEFAULT_DEVPI_LDAPCONFILE=
+
+# http://stackoverflow.com/questions/228544/how-to-tell-if-a-string-is-not-defined-in-a-bash-shell-script
+[[ -z ${DEVPI_SERVERDIR+xxx} ]]   && export DEVPI_SERVERDIR=$DEFAULT_DEVPI_SERVERDIR
+[[ -z ${NGINX_IP+xxx} ]]          && export NGINX_IP=$DEFAULT_NGINX_IP
+[[ -z ${NGINX_PORT+xxx} ]]        && export NGINX_PORT=$DEFAULT_NGINX_PORT
+[[ -z ${DEVPI_SERVERIP+xxx} ]]    && export DEVPI_SERVERIP=$DEFAULT_DEVPI_SERVERIP
+[[ -z ${DEVPI_SERVERPORT+xxx} ]]  && export DEVPI_SERVERPORT=$DEFAULT_DEVPI_SERVERPORT
+[[ -z ${DEVPI_CLIENTDIR+xxx} ]]   && export DEVPI_CLIENTDIR=$DEFAULT_DEVPI_CLIENTDIR
+[[ -z ${DEVPI_CLIENTNAME+xxx} ]]  && export DEVPI_CLIENTNAME=$DEFAULT_DEVPI_CLIENTNAME
+[[ -z ${DEVPI_CLIENTPWD+xxx} ]]   && export DEVPI_CLIENTPWD=$DEFAULT_DEVPI_CLIENTPWD
+[[ -z ${DEVPI_LDAPCONFILE+xxx} ]] && export DEVPI_LDAPCONFILE=$DEFAULT_DEVPI_LDAPCONFILE
+
+#===========================
+# devpi configuration
+# TODO add ldap stuff
+#===========================
+
+# Start the server
+devpi-server --start --host ${DEVPI_SERVERIP} --port ${DEVPI_SERVERPORT}
+
+# Configure a default client if necessary
+if [[ ! -f ${DEVPI_SERVERDIR}/.serverversion ]]; then
+  devpi use http://${DEVPI_SERVERIP}:${DEVPI_SERVERPORT}
+  devpi login ${DEVPI_CLIENTNAME} --password="${DEVPI_CLIENTPWD}"
+  devpi user -m root password="${DEVPI_CLIENTPWD}"
   devpi index -y -c public pypi_whitelist='*'
 fi
-tail -f $DEVPI_SERVERDIR/.xproc/devpi-server/xprocess.log
+
+#===========================
+# log configuration
+# FIXME: devpi-server and nginx logs are mixed
+#===========================
+
+# forward request and error logs to docker log collector
+ln -sf /dev/stdout ${DEVPI_SERVERDIR}/.xproc/devpi-server/xprocess.log
+ln -sf /dev/stdout /var/log/nginx/access.log
+ln -sf /dev/stderr /var/log/nginx/error.log
+
+#===========================
+# nginx configuration
+#===========================
+
+# Edit virtual host based on the environment
+nginx_conf_file=/etc/nginx/sites-available/devpi.conf
+sed -i "s/{{NGINX_IP}}/${NGINX_IP}/g"                 $nginx_conf_file 
+sed -i "s/{{NGINX_PORT}}/${NGINX_PORT}/g"             $nginx_conf_file 
+sed -i "s/{{DEVPI_SERVERPORT}}/${DEVPI_SERVERPORT}/g" $nginx_conf_file 
+# DEVPI_SERVERDIR contains / so it needs to be escaped
+# See http://stackoverflow.com/questions/407523/escape-a-string-for-a-sed-replace-pattern
+sed -i "s/{{DEVPI_SERVERDIR}}/$(echo ${DEVPI_SERVERDIR} | sed -e 's/[\/&]/\\&/g')/g" $nginx_conf_file 
```

Comment:
btw, I have tested my sed rules with slashes in envvar values and it worked fine.